spec/api/search_spec.rb

```diff
@@ -1,33 +1,40 @@
 require 'spec_helper'
 require 'unicode_shared_examples'
 
-describe "app" do
-  describe "search" do
+describe 'app' do
+  describe 'search' do
     include_context 'search_enabled'
 
     before (:each) { set_api_key_header }
-    let(:author) { create_test_user(42) }
-    let(:course_id) { "test/course/id" }
+    let(:author) { create(:user) }
+    let(:course_id) { 'test/course/id' }
 
     def get_result_ids(result)
-      result["collection"].map { |t| t["id"] }
+      result['collection'].map { |t| t['id'] }
     end
 
     describe "GET /api/v1/search/threads" do
-      def assert_empty_response
-        last_response.should be_ok
-        result = parse(last_response.body)
-        result.should == {}
+      shared_examples_for 'response for invalid request' do
+        before (:each) { refresh_es_index }
+        subject { get '/api/v1/search/threads', {course_id: course_id}.merge!(parameters) }
+
+        it { should be_ok }
+        it { should be_an_empty_response }
+      end
+
+      context 'without text parameter' do
+        let(:parameters) { {} }
+        it_behaves_like 'response for invalid request'
       end
 
-      it "returns an empty result if text parameter is missing" do
-        get "/api/v1/search/threads", course_id: course_id
-        assert_empty_response
+      context 'with an invalid sort key' do
```

Comment:
Again, are both sort_key and sort_order valid params?  Do we still support both of them?